src/database/import.ts

```diff
@@ -0,0 +1,172 @@
+      method: "GET",
+      path: this.dbUrl.pathname + ".json",
+      queryParams: { shallow: "true" },
+    });
+
+    if (response.body) {
+      throw new FirebaseError(
+        "Importing is only allowed for an empty location. Delete all data by running " +
+          clc.bold(`firebase database:remove ${this.dbUrl.pathname} --disable-triggers`) +
+          ", then rerun this command.",
+        { exit: 2 }
+      );
+    }
+  }
+
+  private readAndWriteChunks(): Promise<any> {
+    const { dbUrl } = this;
+    const chunkData = this.chunkData.bind(this);
+    const writeChunk = this.writeChunk.bind(this);
+    const getJoinedPath = this.joinPath.bind(this);
+
+    const readChunks = new stream.Transform({ objectMode: true });
+    readChunks._transform = function (chunk: { key: string; value: any }, _, done) {
+      const data = { json: chunk.value, pathname: getJoinedPath(dbUrl.pathname, chunk.key) };
+      const chunkedData = chunkData(data);
+      const chunks = chunkedData.chunks || [data];
+      chunks.forEach((chunk: Data) => this.push(chunk));
+      done();
+    };
+
+    const writeChunks = new stream.Transform({ objectMode: true });
+    writeChunks._transform = async function (chunk: Data, _, done) {
+      const res = await writeChunk(chunk);
+      this.push(res);
+      done();
+    };
+
+    return new Promise((resolve, reject) => {
+      const responses: any[] = [];
+      const pipeline = new Chain([
+        this.inStream,
+        Filter.withParser({
+          filter: this.computeFilterString(this.dataPath) || (() => true),
+          pathSeparator: "/",
+        }),
+        StreamObject.streamObject(),
+      ]);
+      pipeline
+        .on("error", (err: any) =>
+          reject(
+            new FirebaseError(
+              `Invalid data; couldn't parse JSON object, array, or value. ${err.message}`,
+              {
+                original: err,
+                exit: 2,
+              }
+            )
+          )
+        )
+        .pipe(readChunks)
+        .pipe(writeChunks)
+        .on("data", (res: any) => responses.push(res))
+        .on("error", reject)
+        .once("end", () => resolve(responses));
+    });
+  }
+
+  private writeChunk(chunk: Data): Promise<ClientResponse<any>> {
+    return this.limit(() =>
+      this.client.request({
+        method: "PUT",
+        path: chunk.pathname + ".json",
+        body: JSON.stringify(chunk.json),
+        queryParams: this.dbUrl.searchParams,
+      })
+    );
+  }
+
+  private chunkData({ json, pathname }: Data): ChunkedData {
+    if (typeof json === "string" || typeof json === "number" || typeof json === "boolean") {
+      // Leaf node, cannot be chunked
+      return { chunks: null, size: JSON.stringify(json).length };
+    } else {
+      // Children node
+      let size = 2; // {}
+
+      const chunks = [];
+      let hasChunkedChild = false;
+
+      for (const key of Object.keys(json)) {
```

Comment:
Optional - since you are accessing json[key] inside this loop, it would be slightly cleaner imo to use Object.entries instead 

```
for (const [key, val] of Object.entries(json))
```

Reply:
Done.